### ISSUE TITLE
d3m.exceptions.InvalidArgumentValueError: Cannot add step 4.

### ISSUE THREAD

**Issue by sagarshankar** [datamllab/tods#74]
I am attempting to implement an isolation forests pipeline with cross validation, but when I run the pipeline I get error saying that my step 4 where I implement k-folds cross validation cannot be added to the pipeline. I have attached my script for generating this pipeline and I am wondering if there is anything that is currently out of order possibly?

<img width="517" alt="Capture_kfold" src="https://user-images.githubusercontent.com/107258366/176928178-48a3b814-7043-4a91-b853-06cab6b3f945.PNG">

<img width="537" alt="Capture" src="https://user-images.githubusercontent.com/107258366/176927802-5f892e3a-98ec-4131-af6c-11f8ba5b0eab.PNG">

**Comment by lhenry15** [comment]
To add a new primitive, you may want to 1) register the primitive into the [entry.ini](https://github.com/datamllab/tods/blob/master/tods/resources/.entry_points.ini) file, 2) re-install the package with "pip install -e ." to reload the registration. For debugging, you may want to check the "input" of the produce function to make sure the input and the output are consistently as "d3m_dataframe".

**Comment by sagarshankar** [comment]
I get this error in return for some reason as well. Also, I checked the "entry.ini" file and it seems that the K-Fold primitive I am using is in there. I also got this other error as well. Which I am trying to figure out why that would be the case. Any feedback would be great.

raise exceptions.InvalidArgumentValueError("Cannot add step {step_index}.".format(step_index=step.index)) from error

**Comment by sagarshankar** [comment]
Yeah this is the full error. It mentions that the K-Fold_Split seems to require some additional arguments and that seems to be the main cause, but I am not sure why that is.

<img width="516" alt="full_error" src="https://user-images.githubusercontent.com/107258366/176954042-ceabe84d-db28-4bc9-b29d-25f49a420ff0.PNG">

**Comment by sagarshankar** [comment]
So I also noticed in the **"KFoldSplit.py"** file, that there is no function called "produce" which is used in many of the other primitives. There is a function called **"def _get_splits"**, but when I say step_4.add_output('_get_splits), I get the following error as well.
<img width="516" alt="error_2" src="https://user-images.githubusercontent.com/107258366/176961695-7c4f07e0-cb85-4feb-a999-32f8b16366cd.PNG">

**Comment by lhenry15** [comment]
In our backend D3M system, only certain APIs of a primitive (e.g., "produce", "fit", "fit_produce") will be approachable by the pipeline constructor. Therefore, you might want to make sure the produce function is the one performing the functionality.

**Comment by sagarshankar** [comment]
So what exactly is the acceptable API for the K Fold primitive? I have tried "produce", "fit" and "fit_produce" but none of them sem to be working so far
